docs/blocks/blocks.md

```diff
@@ -35,6 +35,43 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+Blocks' textures can be programmatically colored. Many Vanilla blocks utilize this functionality. For instance, color of grass, vines, waterlilys and etc. changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+A block color handler is required to color a block and is an instance of `IBlockColor`, which adds the following method.
+
+```java
+int colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+#### Return Value
+
+This method returns a hex representation of a color in an integer. 
+
+#### Parameters
+
+The `IBlockState`, `IBlockAccess` and `BlockPos` passed to the method make it possible to change color multiplier dynamically, but the method can be called with null `IBlockAccess` and `BlockPos` by `BlockModelRenderer`, when it renders model's brightness.
```

Comment:
In that case I don't think it's worth mentioning specifics. Just note that those parameters *are* nullable, not when.